retinaface/RetinaFace.py

```diff
@@ -270,19 +270,58 @@ def extract_faces(
                 nose = landmarks["nose"]
                 # mouth_right = landmarks["mouth_right"]
                 # mouth_left = landmarks["mouth_left"]
-                facial_img = postprocess.alignment_procedure(facial_img, right_eye, left_eye, nose)
+                facial_img, rotate_angle, rotate_direction = postprocess.alignment_procedure(
+                                                                facial_img, right_eye, left_eye, nose
+                                                            )
 
             if align_first is True and len(obj) == 1:
-                facial_img = extract_faces(
-                    img_path=facial_img,
-                    threshold=threshold,
-                    model=model,
-                    allow_upscaling=allow_upscaling,
-                    expand_face_area=expand_face_area,
-                    align=False,
-                    align_first=False,
-                )[0][:, :, ::-1]
+                facial_area = rotate_facial_area(
+                    facial_area, rotate_angle, rotate_direction, img.shape
+                    )
+                # expand the facial area to be extracted and stay within img.shape limits
+                x1 = max(0, facial_area[0] - int((facial_area[2] * expand_face_area) / 100))
+                y1 = max(0, facial_area[1] - int((facial_area[3] * expand_face_area) / 100))
+                x2 = min(img.shape[1], facial_area[2] + int((facial_area[2] * expand_face_area) / 100))
```

Comment:
linting fails because this line is too long - https://github.com/serengil/retinaface/actions/runs/7857276051/job/21441019715?pr=80

you may consider to make it shorter as

```python
x2 = min(
      img.shape[1], facial_area[2] + int((facial_area[2] * expand_face_area) / 100)
)
```

Reply:
I just updated the line as follows:
```
x2 = min(img.shape[1], facial_area[2] + 
                     int((facial_area[2] * expand_face_area) / 100))
```
Should i change again?

Reply:
Nope, that also works